lib/Sema/ConstraintLocator.h

```diff
@@ -396,19 +396,19 @@ class ConstraintLocator : public llvm::FoldingSetNode {
     /// subclass, returning \c None if unsuccessful.
     template <class T>
     Optional<T> getAs() const {
-      if (auto *result = dyn_cast<T>(this))
+      if (auto *result = llvm::dyn_cast<T>(this))
```

Comment:
Hm, if we don't need these in `TypeBase::getAs` et al, we shouldn't need them here either. Can you look into this further and see what's different about LocatorPathElt?

Reply:
Don't have much time right now, I'll look more extensively later. Right now, the only difference I see is that this is in a nested namespace `swift::constraints` instead of just `swift`. I don't see why this would cause an issue though, so maybe that's not it...

Reply:
Oh, that could be it. We import the LLVM functions into `swift` in include/swift/Basic/LLVM.h; maybe that's treated differently in MSVC.

Reply:
Yeah, MSVC does the lookup in the current explicit namespace.  It’s really annoying that the compilers behave differently there, but IIRC, nothing in the spec prevents that.